src/nemo_run/core/execution/slurm.py

```diff
@@ -879,26 +880,69 @@ def materialize(self) -> str:
                 if group_ind == len(self.slurm_config.resource_group) - 1:
                     srun_command += "\n\nwait\n"
                 else:
-                    srun_command += "\n\nsleep 30\n"
+                    srun_command += f"\n\nsleep {self.slurm_config.wait_time_for_group_job}\n"
                 srun_commands.append(srun_command)
             else:
+                cmd_stdout = srun_stdout.replace(original_job_name, self.jobs[group_ind])
+                cmd_stderr = stderr_flags.copy()
+                if cmd_stderr:
+                    cmd_stderr[-1] = cmd_stderr[-1].replace(original_job_name, self.jobs[group_ind])
+
+                if self.slurm_config.run_as_group and len(self.slurm_config.resource_group) == len(
+                    self.command_groups
+                ):
+                    resource_req = self.slurm_config.resource_group[group_ind]
+                    _container_flags = get_container_flags(
+                        base_mounts=resource_req.container_mounts,
+                        src_job_dir=os.path.join(
+                            slurm_job_dir,
+                            job_directory_name,
+                        ),
+                        container_image=resource_req.container_image,
+                    )
+                    _srun_args = resource_req.srun_args or []
+                    _srun_args = copy.deepcopy(_srun_args)
+                    _srun_args += ["--wait=60", "--kill-on-bad-exit=1"]
+                else:
+                    _container_flags = get_container_flags(
+                        base_mounts=self.slurm_config.container_mounts,
+                        src_job_dir=os.path.join(
+                            slurm_job_dir,
+                            job_directory_name,
+                        ),
+                        container_image=self.slurm_config.container_image,
+                    )
+                    _srun_args = self.slurm_config.srun_args or []
```

Comment:
Can you please move the user-provided arguments to the end of the list if possible? This way I can override the default waiting time. Or make it customizable somehow, as it becomes quite inconvenient to wait for so long if we are debugging something (and we do often debug things on slurm when developing new pipelines)

Reply:
Moved custom srun args to the end